R/causal_impact.R

```diff
@@ -0,0 +1,186 @@
+# Wrapper functions around CausalImpact
+
+#' broom::glance() implementation for bsts (Bayesian Structural Time Series) model,
+#' which is the model used internally in CausalImpact to predict synthetic control (counterfactual).
+#' @export
+glance.bsts <- function(x) {
+  ret <- summary(x)
+  data.frame(residual.sd = ret$residual.sd,
+             prediction.sd = ret$prediction.sd,
+             rsquare = ret$rsquare,
+             relative.gof = ret$relative.gof,
+             coef_min = ret$size[[1]],
+             coef_1st_quartile = ret$size[[2]],
+             coef_median = ret$size[[3]],
+             coef_mean = ret$size[[4]],
+             coef_3rd_quartile = ret$size[[5]],
+             coef_max = ret$size[[6]])
+}
+
+#' broom::tidy() implementation for bsts model
+#' @export
+tidy.bsts <- function(x) {
+  as.data.frame(summary(x)$coefficients)
+}
+
+#' NSE version of do_causal_impact_
+#' @export
+do_causal_impact <- function(df, time, formula, ...) {
+  time_colname <- col_name(substitute(time))
+  do_causal_impact_(df, time_colname, formula = formula, ...)
+}
+
+#' @param df - Data frame
+#' @param time_colname - Column that has time data
```

Comment:
OK, I thought time_colname was a better name, since it is explicit about that it is a name rather than column vector itself, but let's keep it consistent with others for now.